ui/root/src/commonMain/kotlin/social/androiddev/root/splash/DefaultSplashComponent.kt

```diff
@@ -7,15 +7,30 @@
  *
  * You should have received a copy of the GNU General Public License along with Dodo. If not, see <https://www.gnu.org/licenses/>.
  */
-package social.androiddev.root.navigation
+package social.androiddev.root.splash
 
 import com.arkivanov.decompose.ComponentContext
+import com.arkivanov.essenty.instancekeeper.getOrCreate
+import kotlinx.coroutines.flow.StateFlow
+import org.koin.core.component.KoinComponent
+import org.koin.core.component.inject
+import social.androiddev.domain.authentication.usecase.GetSelectedApplicationOAuthToken
+import kotlin.coroutines.CoroutineContext
 
 class DefaultSplashComponent(
     private val componentContext: ComponentContext,
+    private val mainContext: CoroutineContext,
     private val navigateToLandingInternal: () -> Unit,
     private val navigateToTimelineInternal: () -> Unit,
-) : SplashComponent, ComponentContext by componentContext {
+) : SplashComponent, KoinComponent, ComponentContext by componentContext {
+
+    private val getSelectedAuthToken: GetSelectedApplicationOAuthToken by inject()
```

Comment:
I'm thinking about a better way to tell if someone is logged in or not - here I check for the existence of auth token. What do you think?

Reply:
I think we should have a state flow of auth state which is exposed from a repository/use case as a flow. This way you can flatmaplatest (I think that's the switchmap equivalent) and be able to react to auth state changes.  For now I think it would be just login and logout but eventually would encode WHICH user is currently viewing the component when we have multi user management.  IMO the beginning of navigation should be mapped from this flow. Change user and you get new components.

Reply:
Yup, agreed - this should be done at Root component level, which then auth state controls which flow will be taken.
I'm going to address this in this PR